diff -urN screen-3.9.4.orig/configure.in screen-3.9.4/configure.in
--- screen-3.9.4.orig/configure.in	Sat Dec 18 23:03:51 1999
+++ screen-3.9.4/configure.in	Sat Dec 18 23:10:12 1999
@@ -582,8 +582,8 @@
 dnl
 AC_CHECKING(for tgetent)
 olibs="$LIBS"
-LIBS="-lcurses $olibs"
-AC_CHECKING(libcurses)
+LIBS="-Incurses -lncurses $olibs"
+AC_CHECKING(libncurses)
 AC_TRY_LINK(,[
 #ifdef __hpux
 __sorry_hpux_libcurses_is_totally_broken_in_10_10();
@@ -597,10 +597,7 @@
 LIBS="-ltermlib $olibs"
 AC_CHECKING(libtermlib)
 AC_TRY_LINK(,tgetent((char *)0, (char *)0);,,
-LIBS="-lncurses $olibs"
-AC_CHECKING(libncurses)
-AC_TRY_LINK(,tgetent((char *)0, (char *)0);,,
-AC_MSG_ERROR(!!! no tgetent - no screen)))))
+AC_MSG_ERROR(!!! no tgetent - no screen))))
 
 AC_TRY_RUN([
 main()
