--- config.h.in.orig	Fri Sep  1 18:07:26 2000
+++ config.h.in	Fri Sep  8 19:16:57 2000
@@ -88,8 +88,8 @@
  * Note, screen is unable to change mode or group of the pty if it
  * is not installed with sufficient privilege. (e.g. set-uid-root)
  */
-#undef PTYMODE
-#undef PTYGROUP
+#define PTYMODE 0620
+#define PTYGROUP 5 /* RedHat Linux tty */
 
 /*
  * If screen is NOT installed set-uid root, screen can provide tty
