--- Makefile.in.orig	Fri Sep  1 18:07:26 2000
+++ Makefile.in	Fri Sep  8 19:11:49 2000
@@ -68,14 +68,8 @@
 	$(CC) -c -I. -I$(srcdir) $(M_CFLAGS) $(DEFS) $(OPTIONS) $(CFLAGS) $<
 
 install_bin: .version screen
-	-if [ -f $(bindir)/$(SCREEN) ] && [ ! -f $(bindir)/$(SCREEN).old ]; \
-		then mv $(bindir)/$(SCREEN) $(bindir)/$(SCREEN).old; fi
-	$(INSTALL_PROGRAM) screen $(bindir)/$(SCREEN)
-	-chown root $(bindir)/$(SCREEN) && chmod 4755 $(bindir)/$(SCREEN)
-# This doesn't work if $(bindir)/screen is a symlink
-	-if [ -f $(bindir)/screen ] && [ ! -f $(bindir)/screen.old ]; then mv $(bindir)/screen $(bindir)/screen.old; fi
-	rm -f $(bindir)/screen
-	ln -s $(SCREEN) $(bindir)/screen
+	$(INSTALL_PROGRAM) screen $(DESTDIR)$(bindir)/$(SCREEN)
+	-chown root $(bindir)/$(SCREEN) && chmod 4755 $(DESTDIR)$(bindir)/$(SCREEN)
 
 ###############################################################################
 install: installdirs install_bin
@@ -91,7 +85,7 @@
 
 installdirs:
 # Path leading to ETCSCREENRC and Socketdirectory not checked.
-	$(srcdir)/etc/mkinstalldirs $(bindir)
+	$(srcdir)/etc/mkinstalldirs $(DESTDIR)$(bindir)
 	cd doc ; $(MAKE) installdirs
 
 uninstall: .version
