--- screen-4.1/input.c~	2010-02-24 14:51:55.000000000 +0200
+++ screen-4.1/input.c	2010-02-24 14:53:38.919505608 +0200
@@ -38,7 +38,7 @@
 
 struct inpline
 {
-  char  buf[101];	/* text buffer */
+  char  buf[MAX_INPUTLINE];	/* text buffer */
   int  len;		/* length of the editible string */
   int  pos;		/* cursor position in editable string */
 };
@@ -49,7 +49,7 @@
 struct inpdata
 {
   struct inpline inp;
-  int  inpmaxlen;	/* 100, or less, if caller has shorter buffer */
+  int  inpmaxlen;	/* MAX_INPUTLINE-1, or less, if caller has shorter buffer */
   char *inpstring;	/* the prompt */
   int  inpstringlen;	/* length of the prompt */
   int  inpmode;		/* INP_NOECHO, INP_RAW, INP_EVERY */
@@ -133,8 +133,8 @@
   if (!flayer)
     return;
 
-  if (len > 100)
-    len = 100;
+  if (len > MAX_INPUTLINE - 1)
+    len = MAX_INPUTLINE - 1;
   if (!(mode & INP_NOECHO))
     {
       maxlen = flayer->l_width - 1 - strlen(istr);
--- screen-4.1/process.c~	2010-02-24 15:30:55.000000000 +0200
+++ screen-4.1/process.c	2010-02-24 15:32:17.993069465 +0200
@@ -1777,7 +1777,7 @@
       s = *args;
       if (!args[0])
 	{
-	  Input("Stuff:", 100, INP_COOKED, StuffFin, NULL, 0);
+	  Input("Stuff:", MAX_INPUTLINE - 1, INP_COOKED, StuffFin, NULL, 0);
 	  break;
 	}
       n = *argl;
@@ -2063,7 +2063,7 @@
 	ChangeAKA(fore, *args, strlen(*args));
       break;
     case RC_COLON:
-      Input(":", 100, INP_EVERY, Colonfin, NULL, 0);
+      Input(":", MAX_INPUTLINE - 1, INP_EVERY, Colonfin, NULL, 0);
       if (*args && **args)
 	{
 	  s = *args;
@@ -3236,7 +3236,7 @@
 	      Msg(0, "%s: password: window required", rc_name);
 	      break;
 	    }
-	  Input("New screen password:", 100, INP_NOECHO, pass1, display ? (char *)D_user : (char *)users, 0);
+	  Input("New screen password:", MAX_INPUTLINE - 1, INP_NOECHO, pass1, display ? (char *)D_user : (char *)users, 0);
 	}
       break;
 #endif				/* PASSWORD */
@@ -6258,7 +6258,7 @@
     free((char *)u->u_password);
   u->u_password = SaveStr(buf);
   bzero(buf, strlen(buf));
-  Input("Retype new password:", 100, INP_NOECHO, pass2, data, 0);
+  Input("Retype new password:", MAX_INPUTLINE - 1, INP_NOECHO, pass2, data, 0);
 }
 
 static void
--- screen-4.0.2/screen.h~	2006-02-23 00:27:51.000000000 +0200
+++ screen-4.0.2/screen.h	2006-02-23 00:31:32.000000000 +0200
@@ -270,6 +270,7 @@
 #define INP_RAW		2
 #define INP_EVERY	4
 
+#define MAX_INPUTLINE 512
 
 #ifdef MULTIUSER
 struct acl
