diff -urN screen-3.9.15.org/config.h.in screen-3.9.15/config.h.in
--- screen-3.9.15.org/config.h.in	Tue Mar 25 19:06:38 2003
+++ screen-3.9.15/config.h.in	Tue Mar 25 19:07:21 2003
@@ -103,8 +103,9 @@
  * filesystem so screen should not even attempt to set mode or group  
  * even if running as root (e.g. on TiVo).
  */
-#undef PTYMODE
-#undef PTYGROUP
+
+#define PTYMODE 0620
+#define PTYGROUP 5 /* RedHat Linux tty */
 #undef PTYROFS
 
 /*
