--- screen-4.1/configure.in~	2010-03-08 16:29:51.000000000 +0200
+++ screen-4.1/configure.in	2010-03-08 16:30:07.081578872 +0200
@@ -664,6 +656,12 @@
 tgetent((char *)0, (char *)0);
 #endif
 ],,
+LIBS="-ltinfo $olibs"
+AC_CHECKING(libtinfo)
+AC_TRY_LINK(,tgetent((char *)0, (char *)0);,,
+LIBS="-lncurses $olibs"
+AC_CHECKING(libncurses)
+AC_TRY_LINK(,tgetent((char *)0, (char *)0);,,
 LIBS="-ltermcap $olibs"
 AC_CHECKING(libtermcap)
 AC_TRY_LINK(,tgetent((char *)0, (char *)0);,,
@@ -673,10 +671,7 @@
 LIBS="-lncursesw $olibs"
 AC_CHECKING(libncursesw)
 AC_TRY_LINK(,tgetent((char *)0, (char *)0);,,
-LIBS="-lncurses $olibs"
-AC_CHECKING(libncurses)
-AC_TRY_LINK(,tgetent((char *)0, (char *)0);,,
-AC_MSG_ERROR(!!! no tgetent - no screen)))))))
+AC_MSG_ERROR(!!! no tgetent - no screen))))))))
 
 AC_TRY_RUN([
 
