diff -uNr screen-3.7.3.orig/config.h.in screen-3.7.3/config.h.in
--- screen-3.7.3.orig/config.h.in	Mon Aug  4 11:41:12 1997
+++ screen-3.7.3/config.h.in	Mon Aug  4 11:42:53 1997
@@ -77,8 +77,8 @@
  * Note, screen is unable to change mode or group of the pty if it
  * is not installed with sufficient privilege. (e.g. set-uid-root)
  */
-#undef PTYMODE
-#undef PTYGROUP
+#define PTYMODE 0620
+#define PTYGROUP 5 /* RedHat Linux tty */
 
 /*
  * If screen is NOT installed set-uid root, screen can provide tty
