--- screen-4.4.0/doc/screen.texinfo~	2016-06-26 19:27:26.000000000 +0300
+++ screen-4.4.0/doc/screen.texinfo	2016-06-26 19:46:48.088236448 +0300
@@ -3,14 +3,14 @@
 @c vi:set wm=5
 @setfilename screen.info
 @settitle Screen User's Manual
-@dircategory General Commands
 @finalout
 @setchapternewpage odd
 @c %**end of header
 @set version 4.4.0
 
+@dircategory Shell utilities:
 @direntry
-* Screen: (screen).             Full-screen window manager.
+* Screen: (screen).			Full-screen window manager
 @end direntry
 
 @c For examples, use a literal escape in info.
@@ -176,7 +176,7 @@
 variable) who will use it to create the new window.  The above example would 
 start the @code{emacs} editor (editing @file{prog.c}) and switch to its window.
 
-If @file{/etc/utmp} is writable by @code{screen}, an appropriate record
+If @file{/var/run/utmp} is writable by @code{screen}, an appropriate record
 will be written to this file for each window, and removed when the
 window is closed.  This is useful for working with @code{talk},
 @code{script}, @code{shutdown}, @code{rsend}, @code{sccs} and other
@@ -447,7 +447,7 @@
 @cindex screenrc
 When @code{screen} is invoked, it executes initialization commands from
 the files @file{.screenrc} in the user's home directory and
-@file{/usr/local/etc/screenrc}.  These defaults can be overridden in the 
+@file{/etc/screenrc}.  These defaults can be overridden in the 
 following ways:
 For the global screenrc file @code{screen} searches for the environment
 variable @code{$SYSSCREENRC} (this override feature may be disabled at
@@ -968,7 +968,7 @@
 Place, where to dump hardcopy files.  @xref{Hardcopy}.
 @item hardstatus [@var{state}]
 Use the hardware status line.  @xref{Hardware Status Line}.
-@item height [@var{lines} [@var{cols}]]
+@item ns_regheight [@var{lines} [@var{cols}]]
 Set display height.  @xref{Window Size}.
 @item help [-c @var{class}]
 Display current key bindings.  @xref{Help}.
@@ -997,7 +997,7 @@
 @item logfile @var{filename}
 Place where to collect logfiles.  @xref{Log}.
 @item login [@var{state}]
-Log the window in @file{/etc/utmp}.  @xref{Login}.
+Log the window in @file{/var/run/utmp}.  @xref{Login}.
 @item logtstamp [@var{state}]
 Configure logfile time-stamps.  @xref{Log}.
 @item mapdefault
@@ -1561,13 +1561,12 @@
 @kindex C-x
 @deffn Command lockscreen
 (@kbd{C-a x}, @kbd{C-a C-x})@*
-Call a screenlock program (@file{/local/bin/lck} or @file{/usr/bin/lock}
-or a builtin, if no other is available). Screen does not accept any
-command keys until this program terminates. Meanwhile processes in the
-windows may continue, as the windows are in the detached state.
-The screenlock program may be changed through the environment variable
-@code{$LOCKPRG} (which must be set in the shell from which @code{screen}
-is started) and is executed with the user's uid and gid.
+Call a screenlock program (@file{/usr/bin/lck} or a builtin, if no other is
+available). Screen does not accept any command keys until this program
+terminates. Meanwhile processes in the windows may continue, as the windows
+are in the detached state. The screenlock program may be changed through the
+environment variable @code{$LOCKPRG} (which must be set in the shell from
+which @code{screen} is started) and is executed with the user's uid and gid.
 
 Warning: When you leave other shells unlocked and have no password set
 on @code{screen}, the lock is void: One could easily re-attach from an
@@ -1900,7 +1899,7 @@
 * Naming Windows::		Control the name of the window
 * Console::			See the host's console messages
 * Kill::                        Destroy an unwanted window
-* Login::                       Control @file{/etc/utmp} logging
+* Login::                       Control @file{/var/run/utmp} logging
 * Mode::                        Control the file mode of the pty
 * Monitor::                     Watch for activity in a window
 * Windows::			List the active windows
@@ -2109,7 +2108,7 @@
 @kindex L
 @deffn Command login [state]
 (@kbd{C-a L})@*
-Adds or removes the entry in @file{/etc/utmp} for the current window.
+Adds or removes the entry in @file{/var/run/utmp} for the current window.
 This controls whether or not the window is @dfn{logged in}.  In addition
 to this toggle, it is convenient to have ``log in'' and ``log out''
 keys.  For instance, @code{bind I login on} and @code{bind O 
@@ -4962,22 +4961,19 @@
 global initialization files.
 
 @item @code{$SYSSCREENRC}
-@itemx /local/etc/screenrc
+@itemx /etc/screenrc
 @code{screen} initialization commands
 
 @item @code{$SCREENRC}
 @itemx @code{$HOME}/.iscreenrc
 @itemx @code{$HOME}/.screenrc
-Read in after /local/etc/screenrc
+Read in after /etc/screenrc
 
 @item @code{$SCREENDIR}/S-@var{login}
 
-@item /local/screens/S-@var{login}
+@item /tmp/screens/S-@var{login}
 Socket directories (default)
 
-@item /usr/tmp/screens/S-@var{login}
-Alternate socket directories.
-
 @item @var{socket directory}/.termcap
 Written by the @code{dumptermcap} command
 
@@ -4995,7 +4991,7 @@
 @itemx /etc/termcap
 Terminal capability databases
 
-@item /etc/utmp
+@item /var/run/utmp
 Login records
 
 @item @code{$LOCKPRG}
@@ -5100,10 +5096,10 @@
 in order to be able to
 correctly change the owner of the tty device file for each window.
 Special permission may also be required to write the file
-@file{/etc/utmp}.
+@file{/var/run/utmp}.
 
 @item
-Entries in @file{/etc/utmp} are not removed when @code{screen} is killed
+Entries in @file{/var/run/utmp} are not removed when @code{screen} is killed
 with SIGKILL.  This will cause some programs (like "w" or "rwho") to
 advertise that a user is logged on who really isn't.
 
@@ -5184,8 +5180,7 @@
 @cindex socket directory
 
 The socket directory defaults either to @file{$HOME/.screen} or simply to 
-@file{/tmp/screens} or preferably to @file{/usr/local/screens} chosen at 
-compile-time. If @code{screen} is installed
+@file{/tmp/screens} chosen at compile-time. If @code{screen} is installed
 setuid root, then the administrator should compile screen with an
 adequate (not NFS mounted) @code{SOCKDIR}. If @code{screen} is not
 running setuid-root, the user can specify any mode 700 directory in the
