--- config.h.in	2008-06-29 18:04:35.759505981 +0200
+++ config.h.in-new	2008-06-29 18:04:58.139746984 +0200
@@ -564,7 +564,7 @@
  * If you are on a SYS V machine that restricts filename length to 14 
  * characters, you may need to enforce that by setting NAME_MAX to 14
  */
-#undef NAME_MAX		/* KEEP_UNDEF_HERE override system value */
+#undef NAME_MAX		// KEEP_UNDEF_HERE override system value
 #undef NAME_MAX
 
 /*
