--- screen-4.6.2/doc/screen.texinfo.orig	2017-10-23 14:18:26.000000000 +0200
+++ screen-4.6.2/doc/screen.texinfo	2017-12-09 08:56:43.588104192 +0100
@@ -3,14 +3,14 @@
 @c vi:set wm=5
 @setfilename screen.info
 @settitle Screen User's Manual
-@dircategory General Commands
+@dircategory Shell utilities:
 @finalout
 @setchapternewpage odd
 @c %**end of header
 @set version 4.6.2
 
 @direntry
-* Screen: (screen).             Full-screen window manager.
+* Screen: (screen).			Full-screen window manager
 @end direntry
 
 @c For examples, use a literal escape in info.
@@ -184,7 +184,7 @@
 the invoking shell to the application (emacs in this case), because it is
 forked from the parent screen process, not from the invoking shell.
 
-If @file{/etc/utmp} is writable by @code{screen}, an appropriate record
+If @file{/var/run/utmp} is writable by @code{screen}, an appropriate record
 will be written to this file for each window, and removed when the
 window is closed.  This is useful for working with @code{talk},
 @code{script}, @code{shutdown}, @code{rsend}, @code{sccs} and other
@@ -496,7 +496,7 @@
 @cindex screenrc
 When @code{screen} is invoked, it executes initialization commands from
 the files @file{.screenrc} in the user's home directory and
-@file{/usr/local/etc/screenrc}.  These defaults can be overridden in the 
+@file{/etc/screenrc}.  These defaults can be overridden in the 
 following ways:
 For the global screenrc file @code{screen} searches for the environment
 variable @code{$SYSSCREENRC} (this override feature may be disabled at
@@ -1035,7 +1035,7 @@
 Place, where to dump hardcopy files.  @xref{Hardcopy}.
 @item hardstatus [@var{state}]
 Use the hardware status line.  @xref{Hardware Status Line}.
-@item height [@var{lines} [@var{cols}]]
+@item ns_regheight [@var{lines} [@var{cols}]]
 Set display height.  @xref{Window Size}.
 @item help [-c @var{class}]
 Display current key bindings.  @xref{Help}.
@@ -1088,7 +1088,7 @@
 @item logfile @var{filename}
 Place where to collect logfiles.  @xref{Log}.
 @item login [@var{state}]
-Log the window in @file{/etc/utmp}.  @xref{Login}.
+Log the window in @file{/var/run/utmp}.  @xref{Login}.
 @item logtstamp [@var{state}]
 Configure logfile time-stamps.  @xref{Log}.
 @item mapdefault
@@ -1762,13 +1762,12 @@
 @kindex C-x
 @deffn Command lockscreen
 (@kbd{C-a x}, @kbd{C-a C-x})@*
-Call a screenlock program (@file{/local/bin/lck} or @file{/usr/bin/lock}
-or a builtin, if no other is available). Screen does not accept any
-command keys until this program terminates. Meanwhile processes in the
-windows may continue, as the windows are in the detached state.
-The screenlock program may be changed through the environment variable
-@code{$LOCKPRG} (which must be set in the shell from which @code{screen}
-is started) and is executed with the user's uid and gid.
+Call a screenlock program (@file{/usr/bin/lck} or a builtin, if no other is
+available). Screen does not accept any command keys until this program
+terminates. Meanwhile processes in the windows may continue, as the windows
+are in the detached state. The screenlock program may be changed through the
+environment variable @code{$LOCKPRG} (which must be set in the shell from
+which @code{screen} is started) and is executed with the user's uid and gid.
 
 Warning: When you leave other shells unlocked and have no password set
 on @code{screen}, the lock is void: One could easily re-attach from an
@@ -2404,7 +2403,7 @@
 * Naming Windows::		Control the name of the window
 * Console::			See the host's console messages
 * Kill::                        Destroy an unwanted window
-* Login::                       Control @file{/etc/utmp} logging
+* Login::                       Control @file{/var/run/utmp} logging
 * Mode::                        Control the file mode of the pty
 * Monitor::                     Watch for activity or inactivity in a window
 * Windows::			List the active windows
@@ -2613,7 +2612,7 @@
 @kindex L
 @deffn Command login [state]
 (@kbd{C-a L})@*
-Adds or removes the entry in @file{/etc/utmp} for the current window.
+Adds or removes the entry in @file{/var/run/utmp} for the current window.
 This controls whether or not the window is @dfn{logged in}.  In addition
 to this toggle, it is convenient to have ``log in'' and ``log out''
 keys.  For instance, @code{bind I login on} and @code{bind O 
@@ -5802,22 +5801,19 @@
 global initialization files.
 
 @item @code{$SYSSCREENRC}
-@itemx /local/etc/screenrc
+@itemx /etc/screenrc
 @code{screen} initialization commands
 
 @item @code{$SCREENRC}
 @itemx @code{$HOME}/.iscreenrc
 @itemx @code{$HOME}/.screenrc
-Read in after /local/etc/screenrc
+Read in after /etc/screenrc
 
 @item @code{$SCREENDIR}/S-@var{login}
 
-@item /local/screens/S-@var{login}
+@item /tmp/screens/S-@var{login}
 Socket directories (default)
 
-@item /usr/tmp/screens/S-@var{login}
-Alternate socket directories.
-
 @item @var{socket directory}/.termcap
 Written by the @code{dumptermcap} command
 
@@ -5835,7 +5831,7 @@
 @itemx /etc/termcap
 Terminal capability databases
 
-@item /etc/utmp
+@item /var/run/utmp
 Login records
 
 @item @code{$LOCKPRG}
@@ -5860,8 +5856,8 @@
 ============
 
 @verbatim
-     Thomas Renninger <treen@suse.com>,
-     Axel Beckert <abe@deuxchevaux.org>,
+     Thomas Renninger <treen@@suse.com>,
+     Axel Beckert <abe@@deuxchevaux.org>,
      Ken Beal <kbeal@@amber.ssd.csd.harris.com>,
      Rudolf Koenig <rfkoenig@@informatik.uni-erlangen.de>,
      Toerless Eckert <eckert@@informatik.uni-erlangen.de>,
@@ -5945,10 +5941,10 @@
 in order to be able to
 correctly change the owner of the tty device file for each window.
 Special permission may also be required to write the file
-@file{/etc/utmp}.
+@file{/var/run/utmp}.
 
 @item
-Entries in @file{/etc/utmp} are not removed when @code{screen} is killed
+Entries in @file{/var/run/utmp} are not removed when @code{screen} is killed
 with SIGKILL.  This will cause some programs (like "w" or "rwho") to
 advertise that a user is logged on who really isn't.
 
@@ -6028,8 +6024,7 @@
 @cindex socket directory
 
 The socket directory defaults either to @file{$HOME/.screen} or simply to 
-@file{/tmp/screens} or preferably to @file{/usr/local/screens} chosen at 
-compile-time. If @code{screen} is installed
+@file{/tmp/screens} chosen at compile-time. If @code{screen} is installed
 setuid root, then the administrator should compile screen with an
 adequate (not NFS mounted) @code{SOCKDIR}. If @code{screen} is not
 running setuid-root, the user can specify any mode 700 directory in the
