diff -Nru screen-3.9.8/doc/screen.texinfo screen-3.9.8.new/doc/screen.texinfo
--- screen-3.9.8/doc/screen.texinfo	Mon Sep 18 02:24:04 2000
+++ screen-3.9.8.new/doc/screen.texinfo	Mon Sep 18 02:27:19 2000
@@ -790,35 +790,35 @@
 Synonym to @code{acladd}.  @xref{Multiuser Session}.
 @item allpartial @var{state}
 Set all windows to partial refresh.  @xref{Redisplay}.
-@item at @var{[ident][@kbd{#}@var{|}@kbd{*}@var{|}@kbd{%}] command [args]}
+@item at [@var{ident}][@kbd{#}@var{|}@kbd{*}@var{|}@kbd{%}] command [@var{args}]
 Execute a command at other displays or windows.  @xref{At}.
 @item autodetach @var{state}
 Automatically detach the session on SIGHUP.  @xref{Detach}.
 @item autonuke @var{state}
 Enable a clear screen to discard unwritten output.  @xref{Autonuke}.
-@item bce @var{[state]}
+@item bce [@var{state}]
 Change background color erase.  @xref{Character Processing}.
-@item bell_msg @var{[message]}
+@item bell_msg [@var{message}]
 Set the bell notification message.  @xref{Bell}.
-@item bind @var{key [command [args]]}
+@item bind key [@code{command [@var{args}]}]
 Bind a command to a key.  @xref{Bind}.
-@item bindkey @var{[opts] [string [cmd args]]}
+@item bindkey [@code{opts}] [@code{string [@var{cmd args}]}]
 Bind a string to a series of keystrokes. @xref{Bindkey}.
-@item break @var{[duration]}
+@item break [@var{duration}]
 Send a break signal to the current window.  @xref{Break}.
-@item breaktype @var{[tcsendbreak | TCSBRK | TIOCSBRK]}
+@item breaktype [@var{tcsendbreak | TCSBRK | TIOCSBRK}]
 Specify how to generate breaks.  @xref{Break}.
-@item bufferfile @var{[exchange-file]}
+@item bufferfile [@var{exchange-file}]
 Select a file for screen-exchange.  @xref{Screen-Exchange}.
-@item c1 @var{[state]}
+@item c1 [@var{state}]
 Change c1 code processing.  @xref{Character Processing}.
-@item caption @var{mode} @var{[string]}
+@item caption @var{mode} [@var{string}]
 Change caption mode and string.  @xref{Regions}.
 @item chacl @var{usernames permbits list}
 Synonym to @code{aclchg}. @xref{Multiuser Session}.
 @item charset @var{set}
 Change character set slot designation.  @xref{Character Processing}.
-@item chdir @var{[directory]}
+@item chdir [@var{directory}]
 Change the current directory for future windows.  @xref{Chdir}.
 @item clear
 Clear the window screen.  @xref{Clear}.
@@ -842,11 +842,11 @@
 Select default autonuke behavior.  @xref{Autonuke}.
 @item defbce @var{state}
 Select background color erase.  @xref{Character Processing}.
-@item defbreaktype @var{[tcsendbreak | TCSBRK | TIOCSBRK]}
+@item defbreaktype [@var{tcsendbreak | TCSBRK | TIOCSBRK}]
 Specify the default for generating breaks.  @xref{Break}.
 @item defc1 @var{state}
 Select default c1 processing behavior.  @xref{Character Processing}.
-@item defcharset @var{[set]}
+@item defcharset [@var{set}]
 Change defaul character set slot designation.  @xref{Character Processing}.
 @item defescape @var{xy}
 Set the default command and @code{meta} characters.  @xref{Command Character}.
@@ -932,13 +932,13 @@
 Display licensing information.  @xref{Startup}.
 @item lockscreen
 Lock the controlling terminal.  @xref{Lock}.
-@item log @var{[state]}
+@item log [@var{state}]
 Log all output in the current window.  @xref{Log}.
 @item logfile @var{filename}
 Place where to collect logfiles.  @xref{Log}.
-@item login @var{[state]}
+@item login [@var{state}]
 Log the window in @file{/var/run/utmp}.  @xref{Login}.
-@item logtstamp @var{[state]}
+@item logtstamp [@var{state}]
 Configure logfile time-stamps.  @xref{Log}.
 @item mapdefault
 Use only the default mapping table for the next keystroke.  @xref{Bindkey Control}.
@@ -1738,7 +1738,7 @@
 
 @node Resize, Caption, Remove, Regions
 @section Resize
-@deffn Command resize @var{[(+/-)lines]}
+@deffn Command resize [@code{\(+/-\)lines}]
 (none)@*
 Resize the current region. The space will be removed from or added to
 the region below or if there's not enough space from the region above.
@@ -4678,12 +4678,12 @@
 @section Socket Directory
 @cindex socket directory
 
-The socket directory defaults either to @file{$HOME/.screen} or simply to 
-@file{/tmp/screens}} chosen at compile-time. If @code{screen} is installed
-setuid root, then the administrator should compile screen with an
-adequate (not NFS mounted) @code{SOCKDIR}. If @code{screen} is not
-running setuid-root, the user can specify any mode 700 directory in the
-environment variable @code{$SCREENDIR}.
+The socket directory defaults either to @file{$HOME/.screen} or chosen at
+compile-time. If @code{screen} is installed setuid root, then the
+administrator should compile screen with an adequate (not NFS mounted)
+@code{SOCKDIR}. If @code{screen} is not running setuid-root, the user can
+specify any mode 700 directory in the environment variable
+@code{$SCREENDIR}.
 
 @node Compiling Screen,  , Socket Directory, Installation
 @section Compiling Screen
